src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -114,6 +117,28 @@ public Task StartAsync(Uri url, IChannelConnection<Message> application)
             }
         }
 
+        private IEnumerable<Message> ReadMessages(ReadOnlySpan<byte> payload)
+        {
+            if (payload.Length == 0)
+            {
+                yield break;
+            }
+
+            var messageFormat = MessageFormatter.GetFormat(payload[0]);
+            payload = payload.Slice(1);
+
```

Comment:
Shouldn't we throw here if payload is 0? 0 would mean that we received only the messageFromat but not other data...